External/Plugins/HaXeContext/Context.cs

```diff
@@ -1381,10 +1399,60 @@ internal void OnFunctionCompletionResult(HaxeComplete hc, HaxeCompleteStatus sta
 
                 case HaxeCompleteStatus.TYPE:
                     hc.Expr.Position--;
-                    ASComplete.FunctionContextResolved(hc.Sci, hc.Expr, hc.Type, null, true);
+                    ASComplete.FunctionContextResolved(hc.Sci, hc.Expr, result.Type, null, true);
                     break;
             }
         }
+
+        public override bool HandleGotoDeclaration(ScintillaControl sci, ASExpr expression)
+        {
+            if (GetCurrentSDKVersion().IsOlderThan(new SemVer("3.2.0")))
+                return false;
+
+            var hc = new HaxeComplete(sci, expression, false, completionModeHandler, HaxeCompilerService.POSITION);
+            hc.GetPosition(OnPositionCompletionResult);
+            return true;
+        }
+
+        internal void OnPositionCompletionResult(HaxeComplete hc, HaxePositionCompleteResult result, HaxeCompleteStatus status)
+        {
+            switch (status)
```

Comment:
It's ok, but you could have wrapped it all with `InvokeRequired` test and use `TraceManager.Add()` instead of `AddAsync` (that does the same thing).